packages/graphql/src/schema-model/entity/model-adapters/ImplementingEntityOperations.ts

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { upperFirst } from "../../../utils/upper-first";
+import type { InterfaceEntityAdapter } from "./InterfaceEntityAdapter";
+import type { ConcreteEntityAdapter } from "./ConcreteEntityAdapter";
+
+export type RootTypeFieldNames = {
+    create: string;
+    read: string;
+    update: string;
+    delete: string;
+    aggregate: string;
+};
+
+type AggregateTypeNames = {
+    selection: string;
+    input: string;
+};
+
+type MutationResponseTypeNames = {
+    create: string;
+    update: string;
+};
+
+type SubscriptionEvents = {
+    create: string;
+    update: string;
+    delete: string;
+    create_relationship: string;
+    delete_relationship: string;
+};
+
+export type UpdateMutationArgumentNames = {
+    connect: string;
+    disconnect: string;
+    create: string;
+    update: string;
+    delete: string;
+    connectOrCreate: string;
+    where: string;
+};
+
+export type CreateMutationArgumentNames = {
+    input: string;
+};
+
+export class ImplementingEntityOperations<T extends InterfaceEntityAdapter | ConcreteEntityAdapter> {
```

Comment:
So, a question I have here: Conceptually, I'm not sure what `ImplementingEntityOperations` is, I see it is a class to group together the methods that are duplicate between the entityOperations, but I'm not sure if the naming here is ideal.

Any idea on how to make this a bit clearer @Andy2003 ?

Reply:
I used this as a common name for ObjectTypes and InterfaceTypes as it is used in java graphql library (not neo4j-graphql-java):

![image](https://github.com/neo4j/graphql/assets/4464522/90cc9919-94ad-4186-8d7a-05396f551d3f)

In neo4j-graphql-java I use the following names in the domain model:

![image](https://github.com/neo4j/graphql/assets/4464522/7195e727-b152-48f3-870c-e28d98fa3548)

If you have a better name for it, please let me know.

Reply:
I see your point, however, as we have not adopted that convention on this codebase I'm afraid is going to add to the confusion.

I cannot really think of a good name for this, but as this is basically a class to wrap entity operations, it could be simply called `EntityOperations` or `CommonEntityOperations`, maybe even make it abstract if possible

Reply:
But `Entity` would include `Union` which is not the case for these namings. Maybe now is the time to introduce such new "convention"?

Reply:
Maybe it is worth adding, consider at least adding a comment to add some of this context here to avoid confusion down the line, something a bit like

```suggestion
/** ImplementingType refers to the common abstraction of an ObjectType (ConcreteEntity) and InterfaceType */
export class ImplementingEntityOperations<T extends InterfaceEntityAdapter | ConcreteEntityAdapter> {
```